src/demuxer.rs

```diff
@@ -87,49 +83,27 @@ impl Demuxer for Y4MDemuxer {
     }
 }
 
-fn header_token(input: &[u8]) -> IResult<&[u8], &[u8]> {
-    take_till(|c| c == b' ' || c == b'\n')(input)
+fn from_utf8(input: &[u8]) -> std::result::Result<&str, std::str::Utf8Error> {
+    std::str::from_utf8(input)
 }
 
-fn header(input: &[u8]) -> IResult<&[u8], Y4MHeader> {
-    let (mut i, _) = tag("YUV4MPEG2 ")(input)?;
-    let mut width: usize = 0;
-    let mut height: usize = 0;
+fn header_token(input: &[u8]) -> IResult<&[u8], &str> {
+    map_res(take_till(|c| c == b' '), from_utf8)(input).map(|(i, token_str)| {
+        let (_, val) = token_str.split_at(1);
```

Comment:
Always removing the first character does not work for all fields: https://wiki.multimedia.cx/index.php/YUV4MPEG2